lib/ansible/modules/system/sysvinit.py

```diff
@@ -0,0 +1,350 @@
+    returned: always
+    type: complex
+    contains:
+        "attempts": 1
+        "changed": true
+        "name": "apache2"
+        "status": {
+            "enabled": {
+                "changed": true,
+                "rc": 0,
+                "stderr": "",
+                "stdout": ""
+            },
+            "stopped": {
+                "changed": true,
+                "rc": 0,
+                "stderr": "",
+                "stdout": "Stopping web server: apache2.\n"
+            }
+        }
+'''
+
+import re
+from time import sleep
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.service import sysv_is_enabled, get_sysv_script, sysv_exists, fail_if_missing, get_ps, daemonize
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True, type='str', aliases=['service']),
+            state=dict(choices=['started', 'stopped', 'restarted', 'reloaded'], type='str'),
+            enabled=dict(type='bool'),
+            sleep=dict(type='int', default=1),
+            pattern=dict(type='str'),
+            arguments=dict(type='str'),
+            runlevels=dict(type='list'),
+            daemonize=dict(type='bool', default=False),
+        ),
+        supports_check_mode=True,
+        required_one_of=[['state', 'enabled']],
+    )
+
+    name = module.params['name']
+    action = module.params['state']
+    enabled = module.params['enabled']
+    runlevels = module.params['runlevels']
+    pattern = module.params['pattern']
+    sleep_for = module.params['sleep']
+    rc = 0
+    out = err = ''
+    result = {
+        'name': name,
+        'changed': False,
+        'status': {}
+    }
+
+    # ensure service exists, get script name
+    fail_if_missing(module, sysv_exists(name), name)
+    script = get_sysv_script(name)
+
+    # locate binaries for service management
+    paths = ['/sbin', '/usr/sbin', '/bin', '/usr/bin']
+    binaries = ['chkconfig', ' update-rc.d', 'insserv', 'service']
+
+    # Keeps track of the service status for various runlevels because we can
+    # operate on multiple runlevels at once
+    runlevel_status = {}
+
+    location = {}
+    for binary in binaries:
+        location[binary] = module.get_bin_path(binary, opt_dirs=paths)
+
+    # figure out enable status
+    if runlevels:
+        for rl in runlevels:
+            runlevel_status.setdefault(rl, {})
+            runlevel_status[rl]["enabled"] = sysv_is_enabled(name, runlevel=rl)
+    else:
+        runlevel_status["enabled"] = sysv_is_enabled(name)
+
+    # figure out started status, everyone does it different!
+    is_started = False
+
+    # user knows other methods fail and supplied pattern
+    if pattern:
+        is_started = get_ps(module, pattern)
+    else:
+        worked = False
+        if location.get('service'):
+            # standard tool that has been 'destandarized' by reimplementation in other OS/distros
```

Comment:
as standard as the 'ini' format ... it looks the same but its not!

Reply:
@bcoca fortunately, through my testing on devuan and CentOS 6 it appears for the limited use cases covered by the module they do mirror each other's functionality

Reply:
its more 'esoteric' distros or 'non linuces' that worry me, but ... we can wait for feedback as i hope those targets have 'fixed' their divergences

Reply:
@bcoca +1 - I honestly was under the impression that non linux systems completely lacked the service util, but that's probably an erroneous assumption to make

Reply:
i.e freebsd has had 'service' (iirc since 8.0) but it actually uses a rc init system, not sysV.

Reply:
ah +1